services/job_runner.go

```diff
@@ -38,7 +38,7 @@ func BuildRun(job models.Job, store *store.Store) (models.JobRun, error) {
 // ExecuteRun starts the job and executes task runs within that job in the
 // order defined in the run for as long as they do not return errors. Results
 // are saved in the store (db).
-func ExecuteRun(run models.JobRun, store *store.Store, input models.JSON) (models.JobRun, error) {
+func ExecuteRun(run models.JobRun, store *store.Store, input models.RunResult) (models.JobRun, error) {
```

Comment:
Why did we change the last parameter from `JSON` to `RunResult`? Was to ease the passing of the variable from the `JobRunsController` into `resumeJob`? Considering we only use `input.Data`, we should keep this as `JSON`.

Reply:
The main motivation behind that is that the external adapter may want to report an error. That's not yet covered in this PR, but is probably the biggest non-happy-path case I need to cover here. If the job runner takes a full RunResult then we don't have to come up with any funny ways of doing that or designating special keys in the RR's `Data`.